admin/src/js/services/select2-search.js

```diff
@@ -125,15 +154,42 @@ angular.module('inboxServices').factory('Select2Search',
         }
 
         let resolution;
-        let value = selectEl.val();
+        const value = selectEl.val();
         if (!(value && value.length)) {
           resolution = $q.resolve();
         } else {
           if (Array.isArray(value)) {
-            // NB: For now we only support resolving one initial value
-            // multiple is not an existing use case for us
-            value = value[0];
+            // NB: We now support an Array of IDs for places
+            const docPromises = value.map(function (val) { //NoSONAR
+              return getDoc(val).then(function (doc) { //NoSONAR
+                return { id: val, doc: doc };
+              });
+            });
+
+            resolution = $q //NoSONAR
+              .all(docPromises)
+              .then(function(docs) { //NoSONAR
+                const select2Data = selectEl.select2('data') || [];
+                docs.forEach(function (doc) { //NoSONAR
+                  const selected = select2Data.find((d) => d.id === doc.id);
+                  if (selected) {
+                    selected.doc = doc.doc;
+                  } else {
+                    select2Data.push({
+                      id: doc.id,
+                      doc: doc.doc,
+                      text: doc.doc.name,
+                    });
+                  }
+                });
+                selectEl.select2('data', select2Data);
+                $timeout(() => { //NoSONAR
+                  selectEl.trigger('change');
+                }, 1000);
```

Comment:
What is the reason for adding the 1s delay to this and the next timeout?

Reply:
Without it, there was an error from select in the browser console. it seemed like an overlapping of the digest cycles. So I added the small delay.

Reply:
I think the delay is really bad practice. We should find out why the delay is required.